include/ebpf_extension.h

```diff
@@ -18,16 +27,18 @@ typedef ebpf_result_t (*ebpf_program_batch_invoke_function_t)(
     _Out_ uint32_t* result,
     _In_ const void* state);
 
-typedef ebpf_result_t (*ebpf_program_batch_end_invoke_function_t)(_In_ const void* extension_client_binding_context);
-
-typedef ebpf_result_t (*_ebpf_extension_dispatch_function)();
+typedef ebpf_result_t (*ebpf_program_batch_end_invoke_function_t)(
+    _In_ const void* extension_client_binding_context, _Inout_ void* state);
 
-typedef struct _ebpf_extension_dispatch_table
+typedef struct _ebpf_extension_program_dispatch_table
```

Comment:
I think we should name it _hook_ or _link_ (instead of _program_), so as not to confuse with the program info provider NPI.

Reply:
Yep, I wasn't sure what to name these. Is "link" a notion internal to the eBPF implementation? "Hook" is certainly the clearest in intent to a non-eBPF developer.

Reply:
"link" is the name used in libbpf API. It is the same as hook.

Reply:
I've renamed this to `ebpf_link_dispatch_table_t` which I hope is both precise and succinct.